plugin/storage/grpc/factory.go

```diff
@@ -95,19 +95,18 @@ func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger)
 	}
 
 	telset := component.TelemetrySettings{
-		Logger:         logger,
-		TracerProvider: f.tracerProvider,
+		Logger: logger,
 		// TODO needs to be joined with the metricsFactory
 		LeveledMeterProvider: func(_ configtelemetry.Level) metric.MeterProvider {
 			return noopmetric.NewMeterProvider()
 		},
 	}
-	newClientFn := func(opts ...grpc.DialOption) (conn *grpc.ClientConn, err error) {
+	newClientFn := func(telSettings component.TelemetrySettings, opts ...grpc.DialOption) (conn *grpc.ClientConn, err error) {
```

Comment:
Please elaborate on this part. In `newRemoteStorage()` you never modify `telset` when calling `newClientFn` twice, so why do you need to pass it vs. how it was passed already here via closure?